WordPress/Classes/ViewRelated/Gutenberg/EditorMediaUtility.swift

```diff
@@ -165,38 +123,82 @@ class EditorMediaUtility {
         success: @escaping (UIImage) -> Void,
         onFailure failure: @escaping (Error) -> Void
     ) -> ImageDownloaderTask {
+        let postObjectID = post.objectID
+        let result = ContextManager.shared.performQuery { context in
+            Result {
+                try EditorMediaUtility.prepareForDownloading(url: url, size: requestSize, scale: scale, postObjectID: postObjectID, in: context)
+            }
+        }
+
+        let callbackQueue = DispatchQueue.main
+        switch result {
+        case let .failure(error):
+            callbackQueue.async {
+                failure(error)
+            }
+            return EmptyImageDownloaderTask()
+        case let .success((requestURL, mediaHost)):
+            let imageDownload = AuthenticatedImageDownload(
+                url: requestURL,
+                mediaHost: mediaHost,
+                callbackQueue: callbackQueue,
+                onSuccess: success,
+                onFailure: failure
+            )
+            imageDownload.start()
+            return imageDownload
+        }
+    }
+
+    private static func prepareForDownloading(
+        url: URL,
+        size requestSize: CGSize,
+        scale: CGFloat,
+        postObjectID: NSManagedObjectID,
+        in context: NSManagedObjectContext
+    ) throws -> (URL, MediaHost) {
+        // This function is added to debug the issue linked below.
+        let safeExistingObject: (NSManagedObjectID) throws -> NSManagedObject = { objectID in
+            var object: Result<NSManagedObject, Error> = .failure(AuthenticatedImageDownload.DownloadError.blogNotFound)
+            do {
+                // Catch an Objective-C `NSInvalidArgumentException` exception from `existingObject(with:)`.
+                // See https://github.com/wordpress-mobile/WordPress-iOS/issues/20630
+                try WPException.objcTry {
+                    object = Result {
+                        try context.existingObject(with: objectID)
+                    }
+                }
+            } catch {
+                // Send Objective-C exceptions to Sentry for further diagnosis.
+                WordPressAppDelegate.crashLogging?.logError(error)
+                throw error
+            }
+
+            return try object.get()
+        }
+
+        let post = try safeExistingObject(postObjectID) as! Post
```

Comment:
This is neat.

Hopefully we won't have to use it again, but just in case, I wonder how we could make it more discoverable / flexible so other people can reuse it? 🤔

Reply:
I really hope that this PR fixes the issue and makes the extra Objective-C exception become unnecessary. That means we can delete this ugly try catch.

I don't want to move this into a shared places because I don't think we should catch the `NSInvalidArgument` exception, which to me feels like an indication of programming error, rather than Core Data data accessing error.